src/engine/sparqlExpressions/PrefilterExpressionIndex.h

```diff
@@ -80,10 +80,19 @@ class PrefilterExpression {
   // take a look at the actual implementation for derived classes.
   virtual std::unique_ptr<PrefilterExpression> logicalComplement() const = 0;
 
-  // The respective metadata to the blocks is expected to be provided in
-  // a sorted order (w.r.t. the relevant column).
-  std::vector<BlockMetadata> evaluate(const std::vector<BlockMetadata>& input,
-                                      size_t evaluationColumn) const;
+  // Expects that the provided `BlockMetadata` vector adheres to the following
+  // conditions:
+  // (1) unqiueness of blocks
+  // (2) sorted (order)
+  // (3) consistent column values
```

Comment:
```suggestion
  // (3) Constant values for all columns `< evaluationColumn`
  // If `striplIncompleteBlocks == true` then the first and last block may violate condition (3). If so, they are returned as part of the result. (Note: this flag is set to false by internal recursive calls, for external calls, the default value of `true` should be used for convenience, because then the relevant blocks of an index scan can be directly passed to this function.
```